### PR TITLE
Select shaders per Material instead of per Scene

### PR DESCRIPTION
**Pull request by elviscapiaq**

### CODE REVIEW

File: benchmarks/graphics_pipeline/CMakeLists.txt
Comment (Keenuts) [inline benchmarks/graphics_pipeline/CMakeLists.txt:26]: `shader_benchmark_ps_unlit` if you want to use the copy you made, or remove the copy if you want to reuse the other unlit shader.

Comment (elviscapiaq) [reply benchmarks/graphics_pipeline/CMakeLists.txt:26]: Done.

File: benchmarks/graphics_pipeline/main.cpp
Comment (elviscapiaq) [reply benchmarks/graphics_pipeline/main.cpp:819]: Done.

File: src/ppx/knob.cpp
Comment (elviscapiaq) [reply src/ppx/knob.cpp]: Thank you for the explanation. I updated the code to show/hide a knob based on the current scene and never clear them.

File: include/ppx/knob.h
Comment (elviscapiaq) [reply include/ppx/knob.h:1]: Thanks Angela for let us know. 
I think we could need the feature you indicates in the future.

**Comment by elviscapiaq** [comment]
I added a new commit that adds a GPU FPS display. It also adds the GPU work duration, which is the time to render a frame. Please take a look of this new commit.

File: include/ppx/knob.h
Comment (Keenuts) [reply include/ppx/knob.h:1]: I'd be in favor of merging @angela28chen's PR first, since it only adds new API surface, and then this one to use the newly added APIs.

File: benchmarks/graphics_pipeline/main.cpp
Comment (Keenuts) [inline benchmarks/graphics_pipeline/main.cpp:789]: mCurrentSceneIndex

File: benchmarks/graphics_pipeline/main.cpp
Comment (Keenuts) [inline benchmarks/graphics_pipeline/main.cpp:846]: why is this added to this PR?

File: benchmarks/graphics_pipeline/main.cpp
Comment (Keenuts) [inline benchmarks/graphics_pipeline/main.cpp:69]: Add a comment to explain this limit is arbitrary as for now, we don't need more than 5 material IDs.
(Also nit: rename to `kMaxMaterialCount` to explicitly say this is a limit, not the actual count).

**Review by Keenuts — COMMENTED** [review]
Fine with this change, just a few items to fix.
Also, don't add unrelated changes to a PR. 1 PR, 1 feature.
I'd wait for Angela's API change to land before merging this, as Angela's PR adds what you need.

File: benchmarks/graphics_pipeline/main.cpp
Comment (elviscapiaq) [reply benchmarks/graphics_pipeline/main.cpp:69]: Done.

**Comment by elviscapiaq** [comment]
I applied the feedback on the PR and also removed the unrelated commit (Add GPU FPS display) that I added my mistake.

File: benchmarks/graphics_pipeline/main.cpp
Comment (apazylbe) [inline benchmarks/graphics_pipeline/main.cpp:234]: any reason for the extra space after kMaxMaterialCount here, on the next 2 lines?

File: benchmarks/graphics_pipeline/main.cpp
Comment (apazylbe) [inline benchmarks/graphics_pipeline/main.cpp:244]: do we need the extra space after kMaxMaterialCount?

File: benchmarks/graphics_pipeline/main.cpp
Comment (apazylbe) [inline benchmarks/graphics_pipeline/main.cpp:247]: Alternatively you could do the following after the next for loop:
```
// Hide the rest of the knobs
for (size_t i = mScenes[sceneIndex].materials.size(); i < kMaxMaterialCount ; i++) {
      pMaterialsKnobVs[i]->Hide();
      pMaterialsKnobPs[i]->Hide();
}
```

File: benchmarks/graphics_pipeline/main.cpp
Comment (apazylbe) [inline benchmarks/graphics_pipeline/main.cpp:564]: Is it correct in using `data->textures_count`? It shouldn't be using `data->materials_count`?

**Review by apazylbe — APPROVED** [review]

**Comment by elviscapiaq** [comment]
I applied the feedback over the PR and also rebase it on the latest `main` branch.

**Comment by apazylbe** [comment]
Thanks Elvis! By the way, the [contributing](https://github.com/google/bigwheels/blob/main/CONTRIBUTING.md) page suggests to "add new commits to your branch, instead of amending your commit. Adding new commits makes it easier for the reviewer to see what has changed since the last review." Just mentioning for future PRs

**Comment by apazylbe** [comment]
@elviscapiaq Angela's PR that adds knob visibility has been merged, in case you wanted to rebase

**Comment by Keenuts** [comment]
> @elviscapiaq Angela's PR that adds knob visibility has been merged, in case you wanted to rebase

This one might stay on hold until the benchmark workstream is reviewed, to avoid unnecessary work

**Comment by Keenuts** [comment]
Closing this since the benchmark implem is moving in another direction.

**Pull request closed without merging**
